internal/github/auth.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License
+// 2.0; you may not use this file except in compliance with the Elastic License
+// 2.0.
+
+package github
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+const (
+	envAuth       = "GITHUB_TOKEN"
+	authTokenFile = ".elastic/github.token"
+)
+
+// EnsureAuthConfigured method ensures that GitHub auth token is available.
+func (f *AuthTokenFactory) EnsureAuthConfigured() error {
+	if _, err := f.AuthToken(); err != nil {
+		return fmt.Errorf("GitHub authorization token is missing. Please use either environment variable %s or ~/%s: %w",
+			envAuth, authTokenFile, err)
+	}
+
+	return nil
+}
+
+type AuthTokenFactory struct {
```

Comment:
In Golang is preferred to avoid adding "type" descriptive names (like `Factory`) to structs or functions.

```suggestion
type AuthToken struct {
```